source/core_mqtt_agent.c

```diff
@@ -546,8 +560,16 @@ static MQTTStatus_t processCommand( MQTTAgentContext_t * pMqttAgentContext,
     if( pCommand != NULL )
     {
         assert( pCommand->commandType < NUM_COMMANDS );
-        commandFunction = pCommandFunctionTable[ pCommand->commandType ];
-        pCommandArgs = pCommand->pArgs;
+
+        if( ( pCommand->commandType >= NONE ) && ( pCommand->commandType < NUM_COMMANDS ) )
+        {
+            commandFunction = pCommandFunctionTable[ pCommand->commandType ];
+            pCommandArgs = pCommand->pArgs;
+        }
+        else
```

Comment:
should an error be reported here? wouldn't the command being out of bounds of the command types indicate something faulty happened?

Reply:
I added a warning message.